### ISSUE TITLE
[package] glibmm/2.75.0: msvc build fails if all shared

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan-center-index#16833]
### Description

When all dependencies, build of glibmm fails with many unresolved symbols

### Package and Environment Details

* Package Name/Version: **glibmm/2.75.0**
* Operating System+version: **Windows 11**
* Compiler+version: **Visual Studio 2022**
* Conan version: **conan 2.0.2**
* Python version: **Python 3.11.1**

### Conan profile

```jinja
[settings]
os=Windows
arch=x86_64
compiler=msvc
compiler.version=193
compiler.runtime=dynamic
build_type=Release
[options]
[tool_requires]
[conf]
tools.microsoft.msbuild:installation_path=C:/Program Files/Microsoft Visual Studio/2022/Community
[buildenv]
CMAKE_EXPORT_COMPILE_COMMANDS=ON
```

### Steps to reproduce

`conan install --requires="glibmm/2.75.0" -o "*:shared=True" -b missing`

### Logs

[glibmm.log](https://github.com/conan-io/conan-center-index/files/11125761/glibmm.log)

**Comment by LarrxX** [comment]
Yeah, just encountered it myself. @SpaceIm any workaround for now?
